overdrive.py

```diff
@@ -401,8 +401,12 @@ def availability_link_list(self, book_list):
 
         products = book_list['products']
         for product in products:
-            data = dict(id=product['id'],
-                        title=product['title'],
+            if not 'id' in product:
+                cls.log.warn("No ID found in %r", product)
+                continue
+            book_id = product['id']
```

Comment:
Another bit of code further down uses `book_id`, which wasn't previously defined, so I defined it here.